### PR TITLE
Adds ability for Autoloader to load magento module files in controller…

### PR DESCRIPTION
**Pull request by thisisandrew**
…s/ where class name and location conventions are different - needed where a controller class extends a magento module loaded class and relies on the Autoloader to find it

### CODE REVIEW

File: Autoload.php [hunk 1]
```diff
@@ -121,10 +121,36 @@ public function findFile($class)
                 return $dir . DIRECTORY_SEPARATOR . $classFile;
             }
         }
+
+        if ($resolvedPath = $this->findComposerMagentoModuleFile($class)) {
```
Comment (joshdifabio) [inline Autoload.php]: Would it be nicer to simply do:

``` php
return $this->findComposerMagentoModuleFile($class);
```

which would remove an if-statement and reduce complexity a bit?

File: Autoload.php [hunk 2]
```diff
@@ -121,10 +121,36 @@ public function findFile($class)
...
     }
 
+    /**
+     * Allow the autoloader to find composer magento module files in controllers dir
+     * These files don't follow the same class name/location conventions
+     *
+     * @param $class
+     * @return bool|string
+     */
+    private function findComposerMagentoModuleFile($class){
```
Comment (joshdifabio) [inline Autoload.php]: How about renaming this `findMagentoControllerFile`? This functionality is specific to controllers and will work for any Magento module, not only 'Composered' ones?

File: Autoload.php [hunk 3]
```diff
@@ -152,8 +178,9 @@ private function prepareIncludePath()
             }
         }
         
-        set_include_path(implode(PATH_SEPARATOR, $mageAppIncludeDirs));
-        $this->manualIncludeDirs = array_merge($vendorIncludeDirs, $mageLibIncludeDirs, $globalIncludeDirs);
+        $autoIncludeDirs = array_merge($mageAppIncludeDirs, $vendorIncludeDirs, $mageLibIncludeDirs);
+        set_include_path(implode(PATH_SEPARATOR, $autoIncludeDirs));
+        $this->manualIncludeDirs = $globalIncludeDirs;
```
Comment (joshdifabio) [inline Autoload.php:180]: What is the reason for this change?

Comment (thisisandrew) [inline Autoload.php:180]: There is no change to that method it's just git being weird. Don't know why it showed a diff :(
